File: backend/src/main/java/com/pmrs/backend/service/CompanyService.java
package com.pmrs.backend.service;

import com.pmrs.backend.entity.Company;

import java.util.List;

public interface CompanyService {

    List<Company> getAllCompanies();

    Company getCompanyById(Integer id);

    Company saveCompany(Company company);

    Company updateCompany(Integer id, Company company);

    void deleteCompany(Integer id);

    List<Company> getCompaniesByTier(String tier);

    List<Company> getCompaniesBySector(String sector);
}

File: frontend/src/services/companyService.js
import axios from "axios";

const API_URL = "http://localhost:8080/companies";

export const getAllCompanies = () => axios.get(API_URL);

export const getCompanyById = (id) => axios.get(`${API_URL}/${id}`);

export const addCompany = (company) => axios.post(API_URL, company);

export const updateCompany = (id, company) => axios.put(`${API_URL}/${id}`, company);

export const deleteCompany = (id) => axios.delete(`${API_URL}/${id}`);
